### PR TITLE
[SHEPHERD] Add reset_all function for timer

### PR DESCRIPTION
**Pull request by kylesu98**

### CODE REVIEW

**Comment by brandonxxlee** [comment]
Blocked until #479 is merged in and fixes the other travis checks.

**Review by brandonxxlee — DISMISSED** [review]
LGTM

**Review by JHGZhou — APPROVED** [review]
LGTM

**Pull request merged**
